client/securedrop_client/logic.py

```diff
@@ -417,6 +417,9 @@ def __init__(  # type: ignore[no-untyped-def]
         ):
             os.chmod(self.last_sync_filepath, 0o600)
 
+        # Store currently-selected sources
+        self._selected_sources: set[db.Source] | None = None
```

Comment:
Why the explicit `None` rather than the empty set?

```suggestion
        # Store currently-selected sources from on_receive_selected_sources()
        self._selected_sources: set[db.Source]
```

Reply:
If we used this in more places, and if this higher-level code were less coupled to the database types, I would suggest defining an [alias](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#type-aliases) like—

```python
SourceSet = set[db.Source]
```

—but it's probably not worth it.  Just noting it for the future.

Reply:
That's persuasive.  Thanks for explaining that nuance, @rocodes.